gitbook-docs/request_response.md

```diff
@@ -0,0 +1,137 @@
+# Request/Response
+
+## WebSockets and Other Full-duplex Protocols
+
+The exact format of the message for a specific request is defined elsewhere in the specification.
+
+A request should include the `context` when appropriate and must include a client generated `requestId`. The
+`requestId` is a string and it must be a [version 4 UUID](https://tools.ietf.org/html/rfc4122.html#section-4.4). It
+will always be included in any response to the request by the server.
+
+For example, a request to PUT a value:
+
+```json
+{
+  "context": "vessels.self",
+  "requestId": "123345-23232-232323",
+  "put": {
+    "path": "electrical.switches.anchorLight.state",
+    "value": 1
+  }
+}
+```
+
+The server will respond with a message which includes the `requestId`, `correlationId` if provided, and a `state`.
```

Comment:
`correlationId` is never mentioned again. What's it for?

Reply:
That was left over from a previous version of the document. I removed it.